### DISCUSSION TITLE
Proposal to try Dumper for Node.js applications

### DISCUSSION THREAD

**Discussion by thetutlage** [permafrost-dev/node-ray#303] [Ideas]
Hello!

I see you guys have rolled out a custom implementation for pretty printing values using [pretty-format](https://github.com/permafrost-dev/pretty-format), which returns a CLI output and then converts it to HTML using regular expressions.

I have been in the same boat for a while with [object-inspect](https://github.com/poppinss/object-inspect/blob/master/src/PrettyPrintHtml.js) and decided to create something from scratch that is more flexible with the output by separating the parsing and the printing logic into two different cycles.

I recently created [@poppinss/dumper](https://github.com/poppinss/dumper) (inspired by Symfony's VarDumper), which comes with inbuilt HTML and CLI formatters. You can also use its low-level parser API to create tokens and then print them using a custom formatter. You can even write these formatters using a different language (other than JavaScript).

I decided to share it here to see if you would like to give the project a try :)

**Comment by patinthehat** [comment]
@thetutlage this looks like it would make a great addition! The code for pretty printing was repurposed from a jest internal package originally.  I'll definitely look into this. Thanks! 👍

**Reply by thetutlage** [reply]
Sure, feel free to run the [examples](https://github.com/poppinss/dumper/tree/1.x/examples) to see how the output looks like and let me know if you have any questions
